src/QtAVPlayer/qavplayer.cpp

```diff
@@ -462,6 +463,12 @@ void QAVPlayer::seek(qint64 pos)
     if (pos < 0 || (duration() > 0 && pos > duration()))
         return;
 
+    // allow to seek only if media loaded / valid
+    if(d->mediaStatus != QAVPlayer::LoadedMedia && d->mediaStatus != QAVPlayer::EndOfMedia && d->mediaStatus != QAVPlayer::SeekingMedia)
```

Comment:
What would happen if a user wants to seek before it is loaded? 
or after EOS, or

he wants to seek and decided to seek again while it is seeking? Do we need to drop the latest seek?

Reply:
Seeking before loaded is imo senseless, but seeking after reaching OES makes sense (at least in a case of QCTools / dvrescue) - in fact it will require another PR to customize EOS behavior, but what is needed is ability to keep rendering while 'paused' even on reaching EOS and allow seeking from here. 

As for the seeking again while seeking - it should be allowed too (think of fast seeking using position slider) but I can remove this condition for now for simplicity as it might require introducing some extra synchronization.

Reply:
since we do seeking in async, why we need to limit anything here?
even if the media is not loaded, or eos or in seeking, we can set pending position, and wait until demuxer can handle it?

>what is needed is ability to keep rendering while 'paused' even on reaching EOS and allow seeking from here.
when EOS is reached, StoppedState and EndOfMedia,
when you seek while it is EOS, do you want to see a frame visible?
To see it, you would need to pause.

Reply:
>> even if the media is not loaded, or eos or in seeking, we can set pending position, and wait until demuxer can handle it?

okay. If demuxer will *always* result in state change (even if media is invalid) - agree, condition can be removed

Reply:
>> when EOS is reached, StoppedState and EndOfMedia,

yeah, here I'd like some flexibility, like it could be EOS but still paused (if used to be paused). Or even turning to paused state instead of stopping. So some flags or callback controlling behavior on reaching EOS might be needed.

Reply:
>> when you seek while it is EOS, do you want to see a frame visible?

Yep

Reply:
>> To see it, you would need to pause.

Right, but it probably can be done in separate PR to keep things simple.

Reply:
>> to render the frame while seeking, even if it is EOS?

yes